x-pack/plugins/ml/public/application/explorer/explorer_charts/explorer_chart_distribution.js

```diff
@@ -54,14 +58,75 @@ export class ExplorerChartDistribution extends React.Component {
     seriesConfig: PropTypes.object,
     severity: PropTypes.number,
     tooltipService: PropTypes.object.isRequired,
+    cursor$: PropTypes.object,
   };
 
+  constructor(props) {
+    super(props);
+    this.chartScales = undefined;
+    this.cursor = undefined;
+  }
   componentDidMount() {
-    this.renderChart();
+    const chartScales = this.renderChart();
+    this.updateChartScales(chartScales);
```

Comment:
Do we need to wrap the setting of chartScales in a function - couldn't we just set 
this.chartScales = chartScales directly?

Reply:
Initially I thought it's better to separate logic for `updateChartScales()` and `renderChart()` because they are two separate independent actions. But looking at this again, it's a bit verbose, so I'll move the `this.chartScales = chartScales` within `renderChart()` completely to simplify.